From 728b2634cbf661e5303b1e8cb791b909ad4e907c Mon Sep 17 00:00:00 2001
From: "Guillermo A. Amaral" <g@maral.me>
Date: Wed, 10 Jul 2013 22:05:12 -0700
Subject: [PATCH] Declare O_CLOEXEC on systems with an older kernel and/or
 glibc.

Signed-off-by: Guillermo A. Amaral <g@maral.me>
---
 wiringPi/wiringPi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/wiringPi/wiringPi.c b/wiringPi/wiringPi.c
index ba61d9f..2ee23b9 100644
--- a/wiringPi/wiringPi.c
+++ b/wiringPi/wiringPi.c
@@ -77,6 +77,10 @@
 #define	FALSE	(1==2)
 #endif
 
+#ifndef O_CLOEXEC
+#define O_CLOEXEC 02000000
+#endif
+
 // Environment Variables
 
 #define	ENV_DEBUG	"WIRINGPI_DEBUG"
-- 
1.8.1.5

