diff -uNr v4l-utils-0.8.9-orig/utils/v4l2-ctl/v4l2-ctl-common.cpp v4l-utils-0.8.9-new/utils/v4l2-ctl/v4l2-ctl-common.cpp
--- v4l-utils-0.8.9-orig/utils/v4l2-ctl/v4l2-ctl-common.cpp	2012-10-09 19:22:24.000000000 +0300
+++ v4l-utils-0.8.9-new/utils/v4l2-ctl/v4l2-ctl-common.cpp	2014-07-12 17:24:33.023186383 +0300
@@ -485,16 +485,9 @@
 
 static bool parse_next_subopt(char **subs, char **value)
 {
-	static char *const subopts[] = {
-	    NULL
-	};
-	int opt = getsubopt(subs, subopts, value);
+    *value = strdup(*subs);
+    *subs[0] = 0;
 
-	if (value == NULL) {
-		fprintf(stderr, "No value given to suboption <%s>\n",
-				subopts[opt]);
-		return true;
-	}
 	return false;
 }
 
