diff -uNr v4l-utils-1.6.0/utils/v4l2-ctl/v4l2-ctl-common.cpp v4l-utils-1.6.0-uclibc-fix/utils/v4l2-ctl/v4l2-ctl-common.cpp
--- v4l-utils-1.6.0/utils/v4l2-ctl/v4l2-ctl-common.cpp	2014-10-05 16:43:32.000000000 +0300
+++ v4l-utils-1.6.0-uclibc-fix/utils/v4l2-ctl/v4l2-ctl-common.cpp	2014-12-25 01:42:29.548264977 +0200
@@ -655,7 +655,15 @@
 	static char *const subopts[] = {
 	    NULL
 	};
-	int opt = getsubopt(subs, subopts, value);
+
+	/* ugly ucblic getsubopt hack */
+	if (*subs) {
+	    *value = strdup(*subs);
+	    *subs[0] = 0;
+    }
+
+    char *dummy;
+	int opt = getsubopt(subs, subopts, &dummy);
 
 	if (*value == NULL) {
 		fprintf(stderr, "No value given to suboption <%s>\n",
