--- a/src/macro.h
+++ b/src/macro.h
@@ -36,12 +36,6 @@
 #define _GNU_SOURCE
 #endif
 #include <unistd.h>
-static inline int execvpe(const char *file, char *const argv[],
-                          char *const envp[])
-{
-        environ = (char **)envp;
-        return execvp(file, argv);
-}
 #endif
 #define _printf_attr_(a,b) __attribute__ ((format (printf, a, b)))
 #define _sentinel_ __attribute__ ((sentinel))
